website/docs/about_code_generation.mdx

```diff
@@ -55,13 +79,133 @@ This includes but is not limited to:
 - some Riverpod features will be available only with code generation.
 
 At the same time, many applications already use code generation with packages such
-as [Freezed](https://pub.dev/packages/freezed) or [json_serializable](https://pub.dev/packages/json_serializable).  
+as [Freezed](https://pub.dev/packages/freezed) or [json_serializable](https://pub.dev/packages/json_serializable).
 In that case, your project probably is already setup for code generation, and
 using for Riverpod should be simple.
 
+## Provider type
+
+### <span style={RED_STYLE}>[Sync/Async]</span><span style={BLUE_STYLE}>[Stateless/Stateful]</span>Provider
+:::note
+This is just a way of thinking, not actual types. You don’t need to pick a type
+when using code-generation variant anyway. Instead, <span style={RED_STYLE}>Sync/Async</span> is being
+decided by your return type and <span style={BLUE_STYLE}>Stateless/Stateful</span> is being decided
+by whether you need to perform side-effects.
+:::
+
+<table>
+  <colgroup>
+    <col style={{ minWidth: '200px', maxWidth: '200px' }}/>
+    <col style={MAX_WIDTH_380}/>
+    <col style={MAX_WIDTH_380}/>
+  </colgroup>
+  <tr>
+    <th></th>
+    <th style={{textAlign: 'center'}}><span style={BLUE_20_STYLE}>Stateless</span><br/>(Can’t perform side-effects using public methods)</th>
+    <th style={{textAlign: 'center'}}><span style={BLUE_20_STYLE}>Stateful</span><br/>(Can perform side-effects using public methods)</th>
+  </tr>
+  <tr>
+    <td><span style={RED_16_STYLE}>Sync initialization</span></td>
+    <td style={MAX_WIDTH_380}><CodeBlock language="dart">{trimSnippet(syncStateless)}</CodeBlock></td>
+    <td style={MAX_WIDTH_380}><CodeBlock language="dart">{trimSnippet(syncStateful)}</CodeBlock></td>
+  </tr>
+  <tr>
+    <td><span style={RED_16_STYLE}>Async initialization - Returns a Future</span></td>
```

Comment:
I think it makes more sense to only highlight the `async/sync/stateful/stateless` keywords

Reply:
I've tried highlighting sync/async alone, it sounds weird so:
<img width="848" alt="Screenshot 2023-06-14 at 10 14 16 PM" src="https://github.com/rrousselGit/riverpod/assets/70890146/4934e955-f3b7-4e36-a602-bf15db851dd6">
<img width="709" alt="Screenshot 2023-06-14 at 10 15 04 PM" src="https://github.com/rrousselGit/riverpod/assets/70890146/b53591a8-f940-433e-ba19-9cc6d76b6fcb">

Reply:
I think the bi-color variant looks better on the eye.